XCUITests/AuthenticationTest.swift

```diff
@@ -4,88 +4,38 @@
 
 import XCTest
 
-extension String {
-
-    subscript (i: Int) -> Character {
-        return self[self.characters.index(self.startIndex, offsetBy: i)]
-    }
-
-    subscript (i: Int) -> String {
-        return String(self[i] as Character)
-    }
-
-    subscript (r: Range<Int>) -> String {
-        let start = characters.index(startIndex, offsetBy: r.lowerBound)
-        let end = self.index(start, offsetBy: r.upperBound - r.lowerBound)
-        return self[Range(start ..< end)]
-    }
-}
-
 class AuthenticationTest: BaseTestCase {
-    fileprivate func typePasscode(_ passCode: String) {
-        app.keys[passCode[0]].tap()
-        app.keys[passCode[1]].tap()
-        app.keys[passCode[2]].tap()
-        app.keys[passCode[3]].tap()
-        app.keys[passCode[4]].tap()
-        app.keys[passCode[5]].tap()
-    }
-
-    fileprivate func closeAuthenticationManager() {
-        navigator.goto(BrowserTab)
-    }
-
-    fileprivate func disablePasscode(_ passCode: String) {
-        navigator.goto(PasscodeSettings)
-
-        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode Off"].tap()
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode(passCode)
-        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"])
-    }
-
-    fileprivate func enablePasscode(_ passCode: String, interval: String = "Immediately") {
-        let authenticationmanagerSettingstableviewTable = app.tables["AuthenticationManager.settingsTableView"]
 
-        navigator.goto(PasscodeSettings)
-        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"].tap()
-        waitforExistence(app.staticTexts["Enter a passcode"])
-        typePasscode(passCode)
-        waitforExistence(app.staticTexts["Re-enter passcode"])
-        typePasscode(passCode)
-        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode Off"])
+    fileprivate func setInterval(_ interval: String = "Immediately") {
         navigator.goto(PasscodeIntervalSettings)
-        typePasscode(passCode)
+        let table = app.tables["AuthenticationManager.settingsTableView"]
         app.staticTexts[interval].tap()
         navigator.goto(PasscodeSettings)
-        waitforExistence(authenticationmanagerSettingstableviewTable.staticTexts[interval])
+        waitforExistence(table.staticTexts[interval])
     }
 
-    // Sets the passcode and interval (set to immediately)
     func testTurnOnOff() {
-        enablePasscode("133777")
+        navigator.performAction(Action.SetPasscode)
+        setInterval("Immediately")
         XCTAssertTrue(app.staticTexts["Immediately"].exists)
-        disablePasscode("133777")
+        navigator.nowAt(PasscodeSettings)
```

Comment:
Nit: don't think `nowAt` is needed here. If it is needed, it should probably go in `setInterval`.